subsys/debug/Kconfig

```diff
@@ -238,6 +230,16 @@ config SPIN_VALIDATE
 	  enabled. It adds a relatively hefty overhead (about 3k or so) to
 	  kernel code size, don't use on platforms known to be small.
 
+endif # ASSERT
+
+config FORCE_NO_ASSERT
+	bool "Force-disable no assertions"
+	help
+	  This boolean option disables Zephyr assertion testing even
+	  in circumstances (twister) where it is enabled via
+	  CFLAGS and not Kconfig.  Added solely to be able to work
+	  around compiler bugs for specific tests.
+
 config ASSERT_VERBOSE
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/pull/43769#discussion_r826092374

Reply:
it's using __ASSERT_LOC 
https://github.com/zephyrproject-rtos/zephyr/blob/0521aa0fbce835022fbd65d0d438f4a20d5af893/subsys/bluetooth/common/log.h#L46
I don't know if they should have own macro. Benefit of having common is same formatting.

Reply:
We did have own implementation using printk, but we now favor reuse.

Reply:
If we want a own implementation, it will be a copy of whats in __assert.h, I can send my local changes that I am using currently to output assertions, as a PR.

Otherwise, the changes here look good to me.